cf/read_write/netcdf/netcdfwrite.py

```diff
@@ -455,25 +643,399 @@ def _filled_string_array(self, array, fill_value=""):
+    ):
+        """ "Write a non-standard CFA aggregation instruction term variable
+
+        Wites non-standard CFA terms stored as field ancillaries
+
+        .. versionadded:: TODOCFAVER
+
+        :Parameters:
+
+            field: `Field`
+
+            fragment_ncdimensions: `list` of `str`
+
+            aggregated_data: `dict`
+
+        """
+        aggregated_data_attr = []
+        terms = ["location", "file", "address", "format"]
+        for key, field_anc in self.implementation.get_field_ancillaries(
+            field
+        ).items():
+            if not field_anc.data.cfa_get_term():
+                continue
+
+            data = self.implementation.get_data(field_anc, None)
+            if data is None:
+                continue
+
+            # Check that the field ancillary has the same axes as its
+            # parent field, and in the same order.
+            if field.get_data_axes(key) != field.get_data_axes():
+                continue
+
+            # Still here? Then this field ancillary can be represented
+            #             by a non-standard aggregation term.
+
+            # Then transform the data so that it spans the fragment
+            # dimensions, with one value per fragment. If a chunk has
+            # more than one unique value then the fragment's value is
+            # missing data.
+            dx = data.to_dask_array()
+            dx_ind = tuple(range(dx.ndim))
+            out_ind = dx_ind
+            dx = da.blockwise(
+                self._cfa_unique,
+                out_ind,
+                dx,
+                dx_ind,
+                adjust_chunks={i: 1 for i in out_ind},
+                dtype=dx.dtype,
+            )
+
+            # Get the non-standard term name from the field
+            # ancillary's 'id' attribute
+            term = getattr(field_anc, "id", "term")
+            term = term.replace(" ", "_")
+            name = term
+            n = 0
+            while term in terms:
+                n += 1
+                term = f"{name}_{n}"
+
+            terms.append(term)
+
+            # Create the new CFA term variable
+            term_ncvar = self._cfa_write_term_variable(
+                data=type(data)(dx),
+                ncvar=aggregated_data.get(term, f"cfa_{term}"),
+                ncdimensions=fragment_ncdimensions,
+            )
+
+            aggregated_data_attr.append(f"{term}: {term_ncvar}")
+
+        return aggregated_data_attr
+
+    @classmethod
+    def _cfa_unique(cls, a):
+        """Return the unique value of an array.
+
+        If there are multipl unique vales then missing data is
+        returned.
+
+        .. versionadded:: TODOCFAVER
+
+        :Parameters:
+
+             a: `numpy.ndarray`
+                The array.
+
+        :Returns:
+
+            `numpy.ndarray`
+                A size 1 array containing the unique value, or missing
+                data if there is not a unique unique value.
```

Comment:
```suggestion
                data if there is not a unique value.
```

(you can only be so unique!)